pkg/apis/pipeline/v1/pipeline_types.go

```diff
@@ -275,7 +275,7 @@ func (pt *PipelineTask) validateMatrix(ctx context.Context) (errs *apis.FieldErr
 		errs = errs.Also(pt.Matrix.validateCombinationsCount(ctx))
 	}
 	errs = errs.Also(pt.Matrix.validateParameterInOneOfMatrixOrParams(pt.Params))
```

Comment:
summary: can we remove `validateParameterInOneOfMatrixOrParams`?

Reply:
I suggest not removing `validateParameterInOneOfMatrixOrParams`.

`validateParameterInOneOfMatrixOrParams` checks the params against the `pipelineTask.params` and validates that a `pipelineTask.matrix.Params.foobar` is not defined in `pipelineTask.params`. This validation serves its purpose of comparing params in two separate sections which is still necessary. It is necessary because the error being reported is very clear that a param can either exist in a `pipelineTask.params` or `pipelineTask.matrix.params`. What it does not catch is duplicates within `pipelineTask.matrix.Params` since a `set` is created based on `pipelineTask.matrix.Params`. The set will automatically make sure to have just a single occurrence of any parameter.

The duplicate check `validateDuplicateParameters` each per section, helps identify where exactly the duplicates are and `validateParameterInOneOfMatrixOrParams` compares two sections with a clear error message.

So, to summarize, I would keep these checks as is. Let me know if you think different. I am happy to apply more simplification.

Now, what it reminds me of, we do not have a test for checking if `pipelineTask.matrix.include.params` does not exist in `pipelineTask.params`.   

so instead of only iterating over `m.params`, collect params defined in `matrix.include.params`:

```
func (m *Matrix) validateParameterInOneOfMatrixOrParams(params []Param) (errs *apis.FieldError) {
	matrixParameterNames := sets.NewString()
	if m != nil {
		for _, param := range m.Params {
			matrixParameterNames.Insert(param.Name)
		}
		for _, param := range m.Include.Params {
			matrixParameterNames.Insert(param.Name)
		}
	}
	for _, param := range params {
		if matrixParameterNames.Has(param.Name) {
			errs = errs.Also(apis.ErrMultipleOneOf("matrix["+param.Name+"]", "params["+param.Name+"]"))
		}
	}
	return errs
}
```

thoughts?

Reply:
Thank you Priti! Opened a PR for this here to include in next milestone release: https://github.com/tektoncd/pipeline/pull/6349